src/core_plugins/metrics/public/vis/vis_controller.js

```diff
@@ -0,0 +1,49 @@
+import modules from 'ui/modules';
+import 'plugins/timelion/directives/refresh_hack';
+import 'ui/state_management/app_state';
+import '../directives/visualization';
+const app = modules.get('kibana/metrics_vis');
+
+app.controller('MetricsVisController', (
+  $scope,
+  $element,
+  Private,
+  timefilter,
+  getAppState,
+  $location
+) => {
+
+  // If we are in the visualize editor context (and not embedded) we should not
+  // render the visualizations. This is handled by the editor itself.
+  const embedded = $location.search().embed === 'true';
+  if (!embedded && $scope.vis._editableVis) {
+    $scope.displayVis = false;
+    return;
+  } else {
+    $scope.displayVis = true;
+  }
+
+  // We need to watch the app state for changes to the dark theme attribute.
+  $scope.state = getAppState();
+  $scope.$watch('state.options.darkTheme', newValue => {
+    $scope.backgroundColor = newValue ? '#272727' : '#FFF';
+  });
+
+  const queryFilter = Private(require('ui/filter_bar/query_filter'));
+  const fetch = Private(require('../lib/fetch'));
+
+  $scope.model = $scope.vis.params;
+  $scope.$watch('vis.params', fetch($scope));
+
+  // All those need to be consolidated
```

Comment:
is this comment for the `vis.params` watch of all the watches following? I copied this from Timelion since he figured out all the spots that we needed to hook into.